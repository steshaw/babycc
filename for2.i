# 1 "tests/for2.c"
# 1 "<built-in>"
# 1 "<command line>"
# 1 "tests/for2.c"
int main()
{
    int i;
    int j;

    prints("before\n");

    for (i = 0; i < 2; i = i + 1) {
        for (j = 1; j < 11; j = j + 1) {
            printi(j);
            println();
        }
    }
}
