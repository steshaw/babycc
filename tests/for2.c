int main()
{
    int i;
    int j;

    prints("before\n")

    for (i = 0; i < 2; i = i + 1) {
        prints("in first for\n");
        for (j = 1; j < 11; j = j + 1) {
            prints("in second for\n")
            printi(j);
            println();
        }
    }
}
