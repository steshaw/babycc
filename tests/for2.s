	.file "tests/for2.c"
.global main
	.type	main, @function
main:
	pushl	%ebp
	movl	%esp, %ebp
	subl	$8                                , %esp
	movl	$.LC0, %eax
	pushl	%eax
	call	prints
	movl	$0, %eax
	movl	%eax, -4(%ebp)
.L1:
	movl	-4(%ebp), %eax
	pushl	%eax
	movl	$2, %eax
	popl	%ebx
	cmpl	%eax, %ebx
	setl	%bl
	movzbl	%bl, %eax
	cmpl	$0, %eax
	je	.L2
	movl	$.LC1, %eax
	pushl	%eax
	call	prints
	movl	$1, %eax
	movl	%eax, -8(%ebp)
.L3:
	movl	-8(%ebp), %eax
	pushl	%eax
	movl	$11, %eax
	popl	%ebx
	cmpl	%eax, %ebx
	setl	%bl
	movzbl	%bl, %eax
	cmpl	$0, %eax
	je	.L4
	movl	$.LC2, %eax
	pushl	%eax
	call	prints
	movl	-8(%ebp), %eax
	pushl	%eax
	call	printi
	call	println
	movl	-8(%ebp), %eax
	pushl	%eax
	movl	$1, %eax
	popl	%ebx
	addl	%ebx,%eax
	movl	%eax, -8(%ebp)
	jmp	.L3
.L4:
	movl	-4(%ebp), %eax
	pushl	%eax
	movl	$1, %eax
	popl	%ebx
	addl	%ebx,%eax
	movl	%eax, -4(%ebp)
	jmp	.L1
.L2:
	leave
	ret
	.size	main, .-main
	.section	.rodata
.LC0:
	.string	"before\n"
.LC1:
	.string	"in first for\n"
.LC2:
	.string	"in second for\n"
