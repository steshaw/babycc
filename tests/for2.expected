before
in first for
in second for
1
in second for
2
in second for
3
in second for
4
in second for
5
in second for
6
in second for
7
in second for
8
in second for
9
in second for
10
in first for
in second for
1
in second for
2
in second for
3
in second for
4
in second for
5
in second for
6
in second for
7
in second for
8
in second for
9
in second for
10
